src/Pages/Stocks/Stocks.tsx

```diff
@@ -1,12 +1,15 @@
-import React from 'react'
 import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Button } from '@mui/material';
 import { Link } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
-import { addToWatchList, removeFromWatchList } from '../../state/watchlist/watchlistSlice.js'
+import { addToWatchList, removeFromWatchList, Stock, WatchlistState } from '../../state/watchlist/watchlistSlice.js'
 
-function Stocks({stocks}) {
+interface StocksProps {
+  stocks: Stock[];
+}
+
+function Stocks({stocks}: StocksProps) {
 
-  const watchlist = useSelector(state => state.watchlist);
+  const watchlist = useSelector((state: { watchlist: WatchlistState }) => state.watchlist);
```

Comment:
Если у функции есть возможность тип указать через [Generic](https://www.typescriptlang.org/docs/handbook/2/generics.html) то лучше тип передавать через него:

```suggestion
  const watchlist = useSelector<{ watchlist: WatchlistState }>((state) => state.watchlist);
```